### PR TITLE
compass improvements

### PR DESCRIPTION
**Pull request by saddieeiddas**
This PR adds a number of improvements to the compass system.

Changes
======

#### Compass POI System
- Whole POI system has been refactored
- All POI are now stored in the same place
- Compass renders at 100ms intervals with 100ms css easing
- Feels much smoother when moving around and panning camera
- POI providers are able to detect what POI other providers have created.

#### Compass Track
- Updated styles
- Increased default width to `600px`
- Shows `135` degrees instead of `90`
- Can be resized (`width`)

#### Compass Tooltips
- New widget to show tooltips when you hover over POI in compass

#### Warband POI
- Updated styling for warband members
- POI now changes size depending on how far away the member is
- Clicking on the POI targets the warband member
- Hovering over POI shows tooltip with member name & distance
- Targeted warband member POI shows in blue instead of green

#### Friendly Target POI
- Added a POI provider for friendly target
- POI changes size depending on how far away the target is
- Hovering over POI shows tooltip with target name & distance
- Does not show poi for targeting self or warband member

### CODE REVIEW

File: game/hud/src/components/Compass/CompassPOIManager.tsx [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+export interface CompassPOIManagerProps {
+  children: (value: CompassContext) => React.ReactNode;
+  degreesToShow: number;
+}
+
+export interface CompassPOIManagerState {
+  position: Vec3f;
+}
+
+class CompassPOIManager extends React.Component<CompassPOIManagerProps, CompassPOIManagerState> {
```
Comment (codecorsair) [inline game/hud/src/components/Compass/CompassPOIManager.tsx]: implement shouldComponentUpdate here or make it a PureComponent

File: game/hud/src/components/Compass/POI/WarbandMembers.tsx [hunk 2]
```diff
@@ -6,97 +6,231 @@
...
 `;
 
+const MemberPoiCircle = styled('div')`
+  width: 20px;
+  height: 20px;
+  border-radius: 19px;
+  background: rgba(11, 102, 35,.8);
+  box-shadow: 0 0 1px rgba(11, 102, 35,.8);
+  pointer-events: auto;
+` as React.ComponentType<React.DetailedHTMLProps<React.HTMLAttributes<HTMLDivElement>, HTMLDivElement>>;
```
Comment (codecorsair) [inline game/hud/src/components/Compass/POI/WarbandMembers.tsx]: Why the need for the cast?

File: game/hud/src/services/actions/compass.ts [hunk 3]
```diff
@@ -33,6 +33,32 @@ export function getCompassData() {
   };
 }
 
+export function getCompassFacingData() {
+  let facing: number = client.facing % 360;
+  if (facing < 0) {
```
Comment (codecorsair) [inline game/hud/src/services/actions/compass.ts]: facing can't be < 0 here. Looking at the line above, the sign of the result of the remainder operation is the same as the dividend.

Comment (saddieeiddas) [reply game/hud/src/services/actions/compass.ts]: the `client.facing` value is sometimes negative e.g. `-60`... running `-60 % 360` would give `-60`, the facing is then converted to a positive angle  `-60` is same as `300` .

File: game/hud/src/components/Compass/POI/WarbandMembers.tsx (hunk 2 above)
Comment (saddieeiddas) [reply game/hud/src/components/Compass/POI/WarbandMembers.tsx]: removed this

File: game/hud/src/components/Compass/CompassPOIManager.tsx (hunk 1 above)
Comment (saddieeiddas) [reply game/hud/src/components/Compass/CompassPOIManager.tsx]: added `shouldComponentUpdate` for this

**Comment by saddieeiddas** [comment]
@codecorsair i have pushed updates to fix all issues except the `facing < 0` one as that needs to be there.

File: game/hud/src/components/Compass/CompassPOIManager.tsx (hunk 1 above)
Comment (codecorsair) [reply game/hud/src/components/Compass/CompassPOIManager.tsx]: Thanks

File: game/hud/src/services/actions/compass.ts (hunk 3 above)
Comment (codecorsair) [reply game/hud/src/services/actions/compass.ts]: yea, you're right... idk why I was thinking about that backwards

**Pull request merged**